### PR TITLE
File type check on macosx (if file is "dylib")

### PR DESCRIPTION
**Pull request by moljac**
Context: https://github.com/cake-build/cake/issues/3996

Added filetype/extension check.

Error experienced:

```
Error: Bad IL format. 
The format of the file 
    '/Users/runner/work/1/s/tools/Addins/runtime.osx.10.10-x64.runtime.native.System.Security.Cryptography.Apple.4.3.0/runtimes/osx.10.10-x64/native/System.Security.Cryptography.Native.Apple.dylib' 
is invalid. 
An error occurred when executing task Error: .NET CLI: Process returned an error (exit code 1)
```

The error was experienced at intermittent intervals in CI (and local, but not so often) builds for 2 repos:

https://github.com/xamarin/AndroidX
https://github.com/xamarin/GooglePlayServicesComponents

In last few weeks, alsmost all builds fail regularly.

https://github.com/xamarin/GooglePlayServicesComponents/pull/801

### CODE REVIEW

**Comment by moljac** [comment]
@devlead @patriksvensson @gep13 

Hi

Would you be so kind and review this PR, please? We have blocking issue on GPS-FB-MLKit (GooglePlayServicesComponents) both local and CI builds

https://github.com/xamarin/GooglePlayServicesComponents

File: src/Cake.Core/IO/FileExtensions.cs
Comment (devlead) [inline src/Cake.Core/IO/FileExtensions.cs:136]: A switch on extension would be more efficient and readable
```suggestion
            // Is known extension?
            switch (file.Path?.GetExtension().ToLowerInvariant())
            {
                case "dll":
                case "exe":
                case "sys":
                case "tsp":
                case "acm":
                case "ax":
                case "cpl":
                case "drv":
                case "efi":
                case "mui":
                case "ocx":
                case "scr":
                case null:
                    break;

                default:
                    return false;
            }

```

**Review by devlead — CHANGES_REQUESTED** [review]

File: src/Cake.Core/IO/FileExtensions.cs
Comment (moljac) [reply src/Cake.Core/IO/FileExtensions.cs]: Started with `if` only for `dylib`s. While I was waiting for review and periodically checking I decided to add all other extensions.

File: src/Cake.Core/IO/FileExtensions.cs
Comment (devlead) [inline src/Cake.Core/IO/FileExtensions.cs:139]: My bad, extension includes the `.` so this should be
```suggestion
            switch (file.Path?.GetExtension().ToLowerInvariant())
            {
                case ".dll":
                case ".exe":
                case ".sys":
                case ".tsp":
                case ".acm":
                case ".ax":
                case ".cpl":
                case ".drv":
                case ".efi":
                case ".mui":
                case ".ocx":
                case ".scr":
                case null:
                    break;

                default:
                    return false;
            }
```

**Review by devlead — CHANGES_REQUESTED** [review]

**Review by devlead — CHANGES_REQUESTED** [review]
@moljac Other than the missing periods in extensions it LGTM 👍 
Could you please address, rebase against latest upstream develop branch, and squash/fixup into one commit, then I'll take a final look.

**Comment by moljac** [comment]
I cannot squash. Probably because I updated branch with merge commit (I was in a hurry and doing few things in parallel) and forgot to squash before update.

**Review by devlead — APPROVED** [review]
LGTM 👍

**Comment by devlead** [comment]
@moljac your changes have been merged, thanks for your contribution 👍

**Comment by devlead** [comment]
@moljac FYI a pre-release package is now available on our Azure Artifacts feed:
https://dev.azure.com/cake-build/Cake/_artifacts/feed/cake/NuGet/Cake.Tool/overview/3.2.0-alpha0025

**Comment by moljac** [comment]
> @moljac FYI a pre-release package is now available on our Azure Artifacts feed: https://dev.azure.com/cake-build/Cake/_artifacts/feed/cake/NuGet/Cake.Tool/overview/3.2.0-alpha0025

Great! It works locally. 2 GPS-FB-MLKit PRs build nicely.

When is 3.2.0 supposed to be released?

Thanks a lot!

**Comment by moljac** [comment]
Some feedback: 3.2.0 preview fixes the issue. Confimed on 2 local MBPs and on CI (GPS-FB-MLKit repo https://github.com/xamarin/GooglePlayServicesComponents)

thanks

**Pull request merged**
